code/modules/atmospherics/machinery/pipe.dm

```diff
@@ -685,3 +685,114 @@
 
 	var/turf/T = src.loc			// hide if turf is not intact
 	hide(T.intact)
+
+
+/obj/machinery/atmospherics/pipe/quadway
+	icon = 'icons/obj/atmospherics/pipes/manifold_pipe.dmi'
+#ifdef IN_MAP_EDITOR
+	icon_state = "4way-map"
+#else
+	icon_state = "4way"
+#endif
+	name = "pipe 4-way manifold"
+	desc = "A manifold composed of regular pipes"
+	level = UNDERFLOOR
+	volume = 140
+	var/obj/machinery/atmospherics/node1
+	var/obj/machinery/atmospherics/node2
+	var/obj/machinery/atmospherics/node3
+	var/obj/machinery/atmospherics/node4
+
+/obj/machinery/atmospherics/pipe/quadway/overfloor
+	level = OVERFLOOR
+
+/obj/machinery/atmospherics/pipe/quadway/New()
+	..()
+	initialize_directions = NORTH|SOUTH|EAST|WEST
+
+/obj/machinery/atmospherics/pipe/quadway/hide(var/intact)
+	var/hide_pipe = CHECKHIDEPIPE(src)
+	invisibility = hide_pipe ? INVIS_ALWAYS : INVIS_NONE
+	SET_PIPE_UNDERLAY(src.node1, SOUTH, "short", issimplepipe(src.node1) ?  src.node1.color : null, hide_pipe)
+	SET_PIPE_UNDERLAY(src.node2, WEST, "short", issimplepipe(src.node2) ?  src.node2.color : null, hide_pipe)
+	SET_PIPE_UNDERLAY(src.node3, NORTH, "short", issimplepipe(src.node3) ?  src.node3.color : null, hide_pipe)
+	SET_PIPE_UNDERLAY(src.node4, EAST, "short", issimplepipe(src.node4) ?  src.node4.color : null, hide_pipe)
+
+/obj/machinery/atmospherics/pipe/quadway/pipeline_expansion()
+	return list(node1, node2, node3, node4)
+
+/obj/machinery/atmospherics/pipe/quadway/process()
+	..()
+
+	if(!(src.node1 && src.node2 && src.node3 && src.node4))
+		parent.mingle_with_turf(loc, 70)
+
+/obj/machinery/atmospherics/pipe/quadway/disposing()
+	node1?.disconnect(src)
+	node2?.disconnect(src)
+	node3?.disconnect(src)
+	node4?.disconnect(src)
+	parent = null
+	..()
+
+/obj/machinery/atmospherics/pipe/quadway/disconnect(obj/machinery/atmospherics/reference)
+	if(reference == node1)
+		if(istype(node1, /obj/machinery/atmospherics/pipe))
+			if (parent)
+				parent.dispose()
+			parent = null
+		node1 = null
+
+	if(reference == node2)
+		if(istype(node2, /obj/machinery/atmospherics/pipe))
+			if (parent)
+				parent.dispose()
+			parent = null
+		node2 = null
+
+	if(reference == node3)
+		if(istype(node3, /obj/machinery/atmospherics/pipe))
+			if (parent)
+				parent.dispose()
+			parent = null
+		node3 = null
+
+	if(reference == node4)
+		if(istype(node4, /obj/machinery/atmospherics/pipe))
+			if (parent)
+				parent.dispose()
+			parent = null
+		node4 = null
```

Comment:
4 reference checks, but reference can only be one of the nodes? (Meaning could maybe do else ifs)

Also 4 parent checks, could maybe be made into 1?

Reply:
added else if